tensorflow_datasets/scripts/refactoring/refactoring_dataset.py

```diff
@@ -0,0 +1,178 @@
+
+DATASETS_NAMES = {
+  'audio': ['tedlium'],    # will add full audio_datasets list
+  'image': ['abstract_reasoning'],
+  'image_classification': ['binary_alpha_digits']
+}
+
+def create_dirs(dataset_path: str) -> None:
+  """Creates a new `my_dataset` directory"""
+  if not tf.io.gfile.exists(dataset_path):
+    tf.io.gfile.makedirs(os.path.join(dataset_path, 'fake_data'))
+
+
+def copytree(src, dst):
+  """Copy files and folders recursively"""
+  for item in tf.io.gfile.listdir(src):
+    s = os.path.join(src, item)
+    d = os.path.join(dst, item)
+    if tf.io.gfile.isdir(s):
+      if tf.io.gfile.exists(d):
+        tf.io.gfile.rmtree(d)
+      shutil.copytree(s, d)
+    else:
+      if tf.io.gfile.exists(d):
+        tf.io.gfile.remove(d)
+      tf.io.gfile.copy(s, d)
+
+
+def make_init_file(dataset_type, dataset_name, dataset_path) -> None:
+  """Creates a new __init__.py. file"""
+  file_path = os.path.join(dataset_path, '__init__.py')
+  dataset_name_cls = naming.snake_to_camelcase(dataset_name)
+  data = {'dataset_type': dataset_type,
+          'dataset_name': dataset_name,
+          'dataset_name_cls': dataset_name_cls}
+
+  with tf.io.gfile.GFile(file_path, 'w') as f:
+    f.write(_INIT_FILE.format(**data))
+
+
+def copy_checksum_file(src_checksum_path, dest_path) -> None:
+  """Copy checksum.txt file"""
+  if tf.io.gfile.exists(src_checksum_path):
+    tf.io.gfile.copy(src_checksum_path,
+                     os.path.join(dest_path, posixpath.basename(src_checksum_path)),
+                     overwrite=True)
+
+
+def copy_make_data_file(src_fake_data_script_path, dest_path) -> None:
+  """Copy fake data genneration script file"""
+  if tf.io.gfile.exists(src_fake_data_script_path):
+    tf.io.gfile.copy(src_fake_data_script_path,
+                     os.path.join(dest_path, 'make_fake_data.py'),
+                     overwrite=True)
+
+
+def copy_fake_data_dir(src_fake_data_dir, dest_path) -> None:
+  """Copy fake data directory"""
+  if tf.io.gfile.exists(src_fake_data_dir):
+    copytree(src_fake_data_dir, os.path.join(dest_path, 'fake_data'))
+
+
+def copy_dataset_file(src_dataset_path, dest_path) -> None:
+  """Copy my_dataset.py file"""
+  tf.io.gfile.copy(src_dataset_path,
+                   os.path.join(dest_path, posixpath.basename(src_dataset_path)),
+                   overwrite=True)
+
+
+def copy_dataset_test_file(src_dataset_test_path, dest_path) -> None:
+  """Copy my_dataset_test.py file"""
+  tf.io.gfile.copy(src_dataset_test_path,
+                   os.path.join(dest_path, posixpath.basename(src_dataset_test_path)),
+                   overwrite=True)
+
+
+def refactor_dataset(datasets: Dict[str, list]) -> None:
+  """Refactor all the dataset"""
+  for dataset_type, dataset_names in datasets.items():
+    for dataset_name in dataset_names:
+      # Dataset files path
+      fake_data_dir = os.path.join(FAKE_DATA_DIR, dataset_name)
```

Comment:
I feel this should go inside the `copy_fake_data_dir`. Maybe something like:

```py
def copy_fake_data_dir(
    builder_info: BuilderInfo,
    dst_path: pathlin.Path,
) -> None:
```

With `BuilderInfo`:

```py
@dataclasses.dataclass
class BuilderInfo:
  path: pathlib.Path  # Path('.../image/mnist.py')
  type: str  # image, text,...
  name: str  # mnist, ...
  ...


def extract_info(builder_cls: Type[tfds.core.DatasetBuilder]) -> BuilderInfo:
  """Extract path, name,... from the DatasetBuilder class."""
  ...

```

And similarly for other methods.

Reply:
Done